src/ResourceManager/IotHub/Commands.IotHub/IotHub/GetAzureRmIotHubCertificate.cs

```diff
@@ -0,0 +1,89 @@
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+    using ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet(VerbsCommon.Get, "AzureRmIotHubCertificate")]
+    [OutputType(typeof(PSCertificateDescription), typeof(List<PSCertificate>))]
+    public class GetAzureRmIotHubCertificate : IotHubBaseCmdlet
+    {
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Id")]
+        [Alias("Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string CertificateName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ResourceIdParameterSet))
+            {
+                this.ResourceGroupName = IotHubUtils.GetResourceGroupName(this.ResourceId);
+                this.Name = IotHubUtils.GetIotHubName(this.ResourceId);
+            }
+
+            if (!string.IsNullOrEmpty(CertificateName))
+            {
+                CertificateDescription certificateDescription = this.IotHubClient.Certificates.Get(this.ResourceGroupName, this.Name, this.CertificateName);
+                this.WriteObject(IotHubUtils.ToPSCertificateDescription(certificateDescription));
+            }
+            else
+            {
+                CertificateListDescription certificateDescriptions = this.IotHubClient.Certificates.ListByIotHub(this.ResourceGroupName, this.Name);
```

Comment:
It might be nice to check here if you receive only one certificate back and do list format in that case.

Reply:
Done